### PR TITLE
Teleportation wakes & hand-tele portal close time

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

- Magic, cult and bluespace teleportation forms will now leave behind bluespace wakes, which can be follows by activating a hand teleporter on them.
  - You cannot follow teleports to locations such as Reebe, for obvious reason.
  - The chaplain and magic aware personnel are able to see the slipspace wakes left behind.
- Hand teleporters now take time to close the portal, meaning that while the dispel animation is playing, the portal can still be entered.

## Why It's Good For The Game

This allows the hand teleporter to follow targets which are constantly teleporting around in an untracable way, as well as making it harder for the hand teleporter itself to be used to escape any dangerous conflict. This overall reduces the ability for teleportation to be able to get you instantly out of a situation with no follow up actions.

## Testing Photographs and Procedure

https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/6cb4401f-99ee-47a1-9ab0-1143462f83b6

## Changelog
:cl:
balance: The hand teleporter can now lock onto slipspace wakes left behind by bluespace, cult and magic teleportations.
balance: Portals closed by the hand teleporter take 3 seconds to be dispeled. Portals will have an animation upon expiring rather than instantly disappearing.
fix: The hand teleporter will no longer open entry portals inside windows and airlocks
/:cl:

### CODE REVIEW

**Comment by JixS4v** [comment]
Would this also make it so people can reopen a portal to go back to where they were before teleporting into a high risk area?

**Comment by PowerfulBacon** [comment]
Yes. If you have a hand teleporter, you can re-open the portal to a location you were previously at up to 30 seconds after the portal closes.

**Comment by JixS4v** [comment]
I'm sure there's a way to mix up the signals in a fashion that makes it much more difficult for sec to chase you through portals, for example open a portal you don't use, then open another one to the real location. Repeat till you lose sec.

**Comment by PowerfulBacon** [comment]
yea, that would be possible however all of the wakes would be there if you open a portal, and if you directly teleport then it has a guaranteed wake

**Comment by TeomanTheGreat** [comment]
Does this also include the heretics blade break teleport? If so it kinda defeats its purpose as an instant get somewhere safe move (there is a reason you can only ever make two blades and that's because it's capped to prevent people from hoarding blades to instantly run away).

**Comment by JixS4v** [comment]
pretty sure the to a safe place just means on station, nothing related to running away. But I might be wrong.

**Comment by TeomanTheGreat** [comment]
> pretty sure the to a safe place just means on station, nothing related to running away. But I might be wrong.
> 

I believe the whole point of the move is to break your blade (which is inherently risky due to removing your main weapon other than your grasp which is more of a utility) and then teleporting you somewhere else on the station in case you got yourself into a hairy situation.

**Comment by Isy232** [comment]
There's only 2 hand teleporters on the station and they're and extremely rare find from mining/exploration loot.

It gives the item some offensive value instead of merely defensive, an interesting take but I doubt it'll be used all that much as the captain has one of the two and the captain shouldn't be hunting. Still, I don't dislike the feature, especially the use for teleporting back after your portal closes.

The portal close timer I appreciate. Despite being a captain main and escaping numerous antags with the hand tele, it was kind of overpowered. Escaping using it is still very possible, especially if you spam multiple portals so they've no idea which one you took, its more strategic and less trivial now.

File: code/datums/helper_datums/teleport.dm [hunk 1]
```diff
@@ -229,3 +234,28 @@
...
+			continue
+		conflicting_portal.destination = destination
+		return INITIALIZE_HINT_QDEL
+	. = ..()
+	src.destination = destination
+	var/image/I = image(icon = 'icons/effects/effects.dmi', icon_state = "bluestream", layer = HIGH_OBJ_LAYER, loc = src)
+	I.appearance_flags = RESET_ALPHA | RESET_COLOR
+	I.alpha = 255
+	// Holy people can see the wakes
+	add_alt_appearance(/datum/atom_hud/alternate_appearance/basic/blessedAware, "slipspace_wake", I)
```
Comment (itsmeow) [inline code/datums/helper_datums/teleport.dm]: imo it doesn't really make sense for blessed people to see the wake? It should be like.. a diag hud thing

Comment (Pirill) [reply code/datums/helper_datums/teleport.dm]: I like the idea overall but I agree with this, it just feels a little arbitrary for who can see the wakes... Might wanna give it some more thought.

Comment (JixS4v) [reply code/datums/helper_datums/teleport.dm]: probably intended for cult teleports and such, since the chaplain is supposed to directly deal with those.

Comment (PowerfulBacon) [reply code/datums/helper_datums/teleport.dm]: Yea, my idea was that science and magic are somewhat interconnected and that the chaplain can see the cult/magic things for the same reason they can see the wakes.

Comment (PowerfulBacon) [reply code/datums/helper_datums/teleport.dm]: image
Now shows in the diag hud instead

**Comment by Tsar-Salat** [comment]
> * Magic, cult and bluespace teleportation forms will now leave behind bluespace wakes

At work rn, so I can't read the code.

Does this apply to quantum pads, or bluespace pads?

**Comment by Penwin0** [comment]
> Does this also include the heretics blade break teleport? If so it kinda defeats its purpose as an instant get somewhere safe move [there is a reason you can only ever make two blades (3 if using the flesh path which is deemed more difficult) and that's because it's capped to prevent people from hoarding blades to instantly run away].

I think the tools needed for this are specialized enough that It shouldn't be too much of a problem honestly.

**Review by Penwin0 — APPROVED** [review]
Looks flawless to me

**Review by itsmeow — APPROVED** [review]

**Pull request merged**